Assignment-5/CarDrive.ts

```diff
@@ -0,0 +1,131 @@
+import {
+    CarAbrupltyStopException,
+    CollisionAccidentException,
+    HitAndRunAccidentException,
+    EnvironmentalAccidentException,
+    CarBrakeFailException
+  } from './exceptions';
+    
+  try {
+    driveCar();
+  } catch (e) {
+    if (e instanceof CarBrakeFailException) {
+      console.log("Car Brake Fail Exception occurred.");
+      handleBreakFailDamage();
+    } else if (e instanceof CollisionAccidentException) {
+      console.log("Collision Accident Exception occurred.");
+      handleCollisionAccident();
+    } else if (e instanceof EnvironmentalAccidentException) {
+      console.log("Environmental Accident Exception occurred.");
+      handleNatureRelatedAccident();
+    } else if (e instanceof HitAndRunAccidentException) {
+      console.log("Hit and Run Accident Exception occurred.");
+      handleHitAndRunAccident();
+    }else if (e instanceof CarAbrupltyStopException){
+      console.log("Car Abruplty Stop Exception occurred.");
+      AbrupltyException();
+    } else {
+      console.error("An error occurred:", e.message);
+    }
+  } finally {
+    cleanUpAfterAccident();
+  }
+
+  function driveCar() {
+    const isCarBreakFail = true;
+    const isCollision = false;
+    const isEnvironmental = false;
+    const isAbrupltyStop = false;
+  
+    if (isCarBreakFail) {
+      throw new CarBrakeFailException(
+        "Car Break Fail due to NonService of car, Have proper service"
+      );
+    } else if (isCollision) {
+      throw new CollisionAccidentException(
+        "Collision with another vehicle occurred."
+      );
+    } else if (isAbrupltyStop) {
+      throw new CarAbrupltyStopException (
+        "Automatically stop and not starting."
+      );
+    } else if (isEnvironmental) {
+      throw new EnvironmentalAccidentException(
+        "Environmental conditions such as thunderstorm. Drive with caution."
+      );
+    } else {
+      throw new HitAndRunAccidentException("Hit and run incident happened.");
+    }
+  }
+  
+  function handleBreakFailDamage() {
+    callMechanic();
+    callTowTruck();
+  }
+  
+  function AbrupltyException() {
+    callMechanic();
+    callTowTruck();
+    seekShelter();
+  }
+
+  function handleCollisionAccident() {
+    callAmbulance();
+    assessInjuries();
+    fileInsuranceClaim();
+  }
+  
+  function handleHitAndRunAccident() {
+    callAmbulance();
+    inspectDamage();
+    callPolice();
+  }
+  
+  function handleNatureRelatedAccident() {
+    callAmbulance();
```

Comment:
The method **callAmbulance** is common in multiple exceptions, please check if we can move it to finally

Reply:
No , it is only in 3 exceptions and also finally block will execute, irrespective of whether or not an accident occurs, so we can't call ambulance in finally .

Reply:
So in that case, please add the positive scenario condition also where no exception is there and the car reaches the destination safely.